src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1717,6 +1717,17 @@ public long onDiskLength()
         return dfile.onDiskLength;
     }
 
+    public long onDiskComponentsSize()
```

Comment:
as @blambov pointed out: `CompactionTask#buildCompactionCandidatesForAvailableDiskSpace may violate some of the expectations of the strategy.`, so I am leaving `buildCompactionCandidatesForAvailableDiskSpace` as it is